lib/client.ts

```diff
@@ -604,22 +613,54 @@ export class ServiceClient {
 
     const retryErrors: ServiceClientError[] = [];
     return new Promise<ServiceClientResponse>((resolve, reject) => {
+      const timerInitial = performance.now();
       const breaker = this.getCircuitBreaker(params);
-      const retryOperation = operation(opts, (currentAttempt: number) => {
+      const requestsAbortCallbacks: (() => void)[] = [];
+      const retryOperation = operation(opts, scheduledRetry => {
         breaker.run(
           (success: () => void, failure: () => void) => {
+            if (this.options.dropAllRequestsAfter) {
+              const timerLeft =
+                this.options.dropAllRequestsAfter -
+                (performance.now() - timerInitial);
+              if (params.dropRequestAfter) {
+                params.dropRequestAfter = Math.min(
```

Comment:
This `params` object is reused across retries. Is it safe to mutate it? I prefer to clone it for each retry unless its reference equality is used somewhere.